### PR TITLE
Update 6.0.0

### PR DESCRIPTION
**Pull request by SC426**
[Upstream](https://github.com/python/importlib_metadata)
[Changelog](https://github.com/python/importlib_metadata/blob/main/CHANGES.rst)

### Actions
- Update version number and sha256
- Change Python version skip to `<38`
- Update dependencies
- Linter fixes
- Add `build_base.bat` for Windows builds

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
I believe that for the host section we are trying to use exact pinings as much as possible

https://github.com/AnacondaRecipes/importlib_metadata-feedstock/blob/85af8f8c8478cb933259b92823c64cccb76c07f4/recipe/meta.yaml#L23&L24

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
 outputs:
   - name: importlib-metadata
-    script: build_base.sh
+    script: build_base.sh  # [unix]
+    script: build_base.bat  # [win]
     requirements:
       host:
         - python
         - pip
         - setuptools >=56
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
        - setuptools
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
-        - setuptools_scm >=3.4.1,!=6.1.0
+        - setuptools_scm >=3.4.1
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
        - setuptools_scm
```

**Review by cbouss — COMMENTED** [review]
Good to go, with the change to setuptools and setuptools_scm

**Review by cbouss — APPROVED** [review]

**Review by ELundby45 — APPROVED** [review]

**Pull request merged**